### PR TITLE
Use go 1.18 to build artifacts

### PR DESCRIPTION
**Pull request by jayofdoom**
This should not merge until we are ready for all armada consumers to use go 1.18.

Issue #1142

### CODE REVIEW

**Review by kannon92 — COMMENTED** [review]
Do you need to change the version in go.sum or go.mod?

**Comment by jayofdoom** [comment]
Nice catch; yes I do. Probably should run `go mod tidy`, too. 

Gonna be honest; I didn't expect this change to pass tests so I didn't jump through all the needed hoops yet :laughing:

**Comment by jayofdoom** [comment]
Once https://github.com/G-Research/armada/pull/1276 merges, this rebased on top of it should pass tests.

File: .circleci/config.yml
Comment (kannon92) [inline .circleci/config.yml:112]: This seems like a large change?  Should we pin ubuntu to a version?

**Review by kannon92 — DISMISSED** [review]

File: .circleci/config.yml
Comment (jayofdoom) [reply .circleci/config.yml:112]: This is the way 1.18 tests are running already -- it'd be pretty stunning to see ubuntu break the most basic of functionality we use in a security update release, and I'd rather get the security updates. I understand why some folks may not wanna do that though.

**Comment by jayofdoom** [comment]
This should not merge without explicit approval and timing from GR.

File: .circleci/config.yml
Comment (kannon92) [reply .circleci/config.yml:112]: Yea for this it is probably fine.

**Comment by d80tb7** [comment]
LGTM- let's coordinate with @JamesMurkin to switch the ci over to requiring 1.18 to pass

**Comment by jayofdoom** [comment]
The work has already been done on the public side; if there's anything I can do to help with other CI systems please just ask.

**Pull request merged**
